File: assertion-roulette/python-unittest/OrganizationUnitsControllerTests_Should_Get_OrganizationUnit_List_Async.py
import unittest
from unittest.mock import Mock

class TestOrganizationUnitsController(unittest.TestCase):
    def test_should_get_organization_unit_list(self):
        organization_unit_app_service_mock = Mock()
        organization_unit_app_service_mock.get_list_async.return_value = PagedListResult(
            items=[
                OrganizationUnitListOutput(name="test_organizationUnit_1", id=str(uuid.uuid4())),
                OrganizationUnitListOutput(name="test_organizationUnit_2", id=str(uuid.uuid4())),
                OrganizationUnitListOutput(name="test_organizationUnit_3", id=str(uuid.uuid4())),
                OrganizationUnitListOutput(name="test_organizationUnit_4", id=str(uuid.uuid4())),
                OrganizationUnitListOutput(name="test_organizationUnit_5", id=str(uuid.uuid4()))
            ],
            total_count=10
        )
        
        organization_units_controller = OrganizationUnitsController(organization_unit_app_service_mock)
        action_result = organization_units_controller.get_organization_units(PagedListInput())
        
        ok_object_result = self.assertIsInstance(action_result.result, OkObjectResult)
        organization_unit_paged_list_result = self.assertIsInstance(ok_object_result.value, PagedListResult)
        
        self.assertEqual(200, ok_object_result.status_code)
        self.assertEqual(10, organization_unit_paged_list_result.total_count)
        self.assertEqual(5, len(organization_unit_paged_list_result.items))

File: assertion-roulette/CSharpXUnit/CSharpXUnit.Tests/Tests/OrganizationUnitsControllerTests_Should_Get_OrganizationUnit_List_Async.cs
using Xunit;

namespace BackOfficeBase.Tests.Web.Api
{
    public class OrganizationUnitsControllerTests
    {
        [Fact]
        public void Should_Get_OrganizationUnit_List_Async()
        {
            {
                var organizationUnitAppServiceMock = new Mock<IOrganizationUnitAppService>();
                organizationUnitAppServiceMock.Setup(x => x.GetListAsync(It.IsAny<PagedListInput>())).ReturnsAsync(new PagedListResult<OrganizationUnitListOutput> { Items = new List<OrganizationUnitListOutput> { new OrganizationUnitListOutput { Name = "test_organizationUnit_1", Id = Guid.NewGuid() }, new OrganizationUnitListOutput { Name = "test_organizationUnit_2", Id = Guid.NewGuid() }, new OrganizationUnitListOutput { Name = "test_organizationUnit_3", Id = Guid.NewGuid() }, new OrganizationUnitListOutput { Name = "test_organizationUnit_4", Id = Guid.NewGuid() }, new OrganizationUnitListOutput { Name = "test_organizationUnit_5", Id = Guid.NewGuid() } }, TotalCount = 10 });
                var organizationUnitsController = new OrganizationUnitsController(organizationUnitAppServiceMock.Object);
                var actionResult = await organizationUnitsController.GetOrganizationUnits(new PagedListInput());
                var okObjectResult = Assert.IsType<OkObjectResult>(actionResult.Result);
                var organizationUnitPagedListResult = Assert.IsType<PagedListResult<OrganizationUnitListOutput>>(okObjectResult.Value);
                Assert.Equal((int)HttpStatusCode.OK, okObjectResult.StatusCode);
                Assert.Equal(10, organizationUnitPagedListResult.TotalCount);
                Assert.Equal(5, organizationUnitPagedListResult.Items.Count());
            }
        }
    }
}
